File: my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::placeholders;
 
class SmartphoneNode : public rclcpp::Node 
{
    public:
        SmartphoneNode() : Node("smartphone") 
        {
            subscriber_ = this->create_subscription<example_interfaces::msg::String>(
                "robot_news", 10, std::bind(&SmartphoneNode::calbackRobotNews, this, _1)
            );
            RCLCPP_INFO(this->get_logger(), "Smartphone Node has been created successfully.");
        }
    
    private:
        void calbackRobotNews(const example_interfaces::msg::String::SharedPtr msg) 
        {
            RCLCPP_INFO(this->get_logger(), "Received news: '%s'", msg->data.c_str());
        }   
        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};
    
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartphoneNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
 
 
class SmartphoneNode(Node): 
    def __init__(self):
        super().__init__("smartphone")
        self.subscriber_  = self.create_subscription(
            String, "robot_news", self.callback_robot_news, 10
        )
        self.get_logger().info("Smartphone Node has been started.")
        
    def callback_robot_news(self, msg):
        self.get_logger().info(f"Received news: {msg.data}")
        # Here you can add more functionality to process the news message
 
 
def main(args=None):
    rclpy.init(args=args)
    node = SmartphoneNode()
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()